test/suites/scheduling/suite_test.go

```diff
@@ -0,0 +1,514 @@
+			env.ExpectCreatedNodeCount("==", 1)
+		})
+		It("should support well-known deprecated labels", func() {
+			nodeSelector := map[string]string{
+				// Deprecated Labels
+				corev1.LabelFailureDomainBetaRegion: env.Region,
+				corev1.LabelFailureDomainBetaZone:   fmt.Sprintf("%s-1", env.Region),
+				"topology.disk.csi.azure.com/zone":  fmt.Sprintf("%s-1", env.Region),
+				"beta.kubernetes.io/arch":           "amd64",
+				"beta.kubernetes.io/os":             "linux",
+				corev1.LabelInstanceType:            "Standard_D4s_v5",
+			}
+			selectors.Insert(lo.Keys(nodeSelector)...) // Add node selector keys to selectors used in testing to ensure we test all labels
+			requirements := lo.MapToSlice(nodeSelector, func(key string, value string) corev1.NodeSelectorRequirement {
+				return corev1.NodeSelectorRequirement{Key: key, Operator: corev1.NodeSelectorOpIn, Values: []string{value}}
+			})
+			deployment := test.Deployment(test.DeploymentOptions{Replicas: 1, PodOptions: test.PodOptions{
+				NodeSelector:     nodeSelector,
+				NodePreferences:  requirements,
+				NodeRequirements: requirements,
+			}})
+			env.ExpectCreated(nodeClass, nodePool, deployment)
+			env.EventuallyExpectHealthyPodCount(labels.SelectorFromSet(deployment.Spec.Selector.MatchLabels), int(*deployment.Spec.Replicas))
+			env.ExpectCreatedNodeCount("==", 1)
+		})
+		It("should support well-known labels for topology and architecture", func() {
+			nodeSelector := map[string]string{
+				// Well Known
+				karpv1.NodePoolLabelKey:     nodePool.Name,
+				corev1.LabelTopologyRegion:  env.Region,
+				corev1.LabelTopologyZone:    fmt.Sprintf("%s-1", env.Region),
+				corev1.LabelOSStable:        "linux",
+				corev1.LabelArchStable:      "amd64",
+				karpv1.CapacityTypeLabelKey: karpv1.CapacityTypeOnDemand,
+			}
+			selectors.Insert(lo.Keys(nodeSelector)...) // Add node selector keys to selectors used in testing to ensure we test all labels
+			requirements := lo.MapToSlice(nodeSelector, func(key string, value string) corev1.NodeSelectorRequirement {
+				return corev1.NodeSelectorRequirement{Key: key, Operator: corev1.NodeSelectorOpIn, Values: []string{value}}
+			})
+			deployment := test.Deployment(test.DeploymentOptions{Replicas: 1, PodOptions: test.PodOptions{
+				NodeSelector:     nodeSelector,
+				NodePreferences:  requirements,
+				NodeRequirements: requirements,
+			}})
+			env.ExpectCreated(nodeClass, nodePool, deployment)
+			env.EventuallyExpectHealthyPodCount(labels.SelectorFromSet(deployment.Spec.Selector.MatchLabels), int(*deployment.Spec.Replicas))
+			env.ExpectCreatedNodeCount("==", 1)
+		})
+		// disabled until we test on subscription with GPU quota
+		PIt("should support well-known labels for a gpu (nvidia)", func() {
+			nodeSelector := map[string]string{
+				v1alpha2.LabelSKUAccelerator:     "A100",
+				v1alpha2.LabelSKUGPUName:         "A100",
+				v1alpha2.LabelSKUGPUManufacturer: "nvidia",
+				v1alpha2.LabelSKUGPUCount:        "1",
+			}
```

Comment:
Do we need to test for A100 or can we run this on a NC6 sku for example?

Reply:
Our e2es run on a sub with GPU quota, just not A100 quota

Reply:
Any GPU would do. Let me give it a try ...